### PR TITLE
Update design of Create Project UI

### PR DESCRIPTION
**Pull request by Steffengreiner**
**What was changed** 
This PR updates the UI to fit the [figma ](https://www.figma.com/file/mYcW5viu5GpPZ852VHYBrV/Start-Page?node-id=861%3A20351)prototype outlined in [DM-530](https://qbicsoftware.atlassian.net/browse/DM-530).

**Technical Information** 
This PR aims to achieve multiple things: 
1.) Introduce a reusable CardLayout with an immutable card design
2.) The CardLayout allows for adding components into its individual sublayouts as desired by the user. 
3.) The CardLayout allows for style changes to its Sublayouts as required. 
4.) CSS definition seperation between the basic CardLayout and the styling of the added components by the CreateProjectLayout

**How to Test**  
1.) Login to the data-manager-app
2.) Click the Create Button on the top left
3.) Click the "blank" button in the dialogue window
4) Observe the new fancy card design layout 

**Possible Review Questions** 
Should CSS stylings be imported on a global level(import directly into styles.css)or as done in this PR via the Cssimporter selector in the component? 
Are the individual style change methods(e.g. removeTitleStyles) really necessary for the CardLayout?

Data-manager-application-project-creation-picture

### CODE REVIEW

**Comment by sven1103** [comment]
Dear @Steffengreiner 
thanks a lot for your contribution.

When I run the app, I get the following result, even with a private window:
image

image

@jenniferboedker need to give here feedback to the overall UI result though

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/components/CardLayout.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    fieldLayout.removeClassNames(currentFieldLayoutStyles);
+  }
+
+  /**
+   * Removes all styling of the titleLayout within the CardLayout
+   */
+  public void removeAllTitleStyles() {
+    String[] currentTitleLayoutStyles = titleLayout.getClassNames().toArray(String[]::new);
+    titleLayout.removeClassNames(currentTitleLayoutStyles);
+  }
```
Comment (KochTobi) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/components/CardLayout.java:241]: What does this public style handling do? It is not used and I do not understand the API. Which buttons are styled in which way?

Comment (jenniferboedker) [reply vaadinfrontend/src/main/java/life/qbic/datamanager/views/components/CardLayout.java:241]: Hi @KochTobi I think this is what steffen mentioned in the PR description. It allows to customize the style of the card and e.g remove styles applied to the sublayouts. E.g the title, the buttons or the content (called field layout here)

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/projectOverview/components/CreationModeDialog.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ private void setupButtonLayout(){
 
     private void setupContent(){
         createCreationModeButtons();
-        Label blankLabel = new Label("From Offer");
+        Label blankLabel = new Label("Blank");
```
Comment (jenniferboedker) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/projectOverview/components/CreationModeDialog.java:50]: 🎉

**Review by jenniferboedker — COMMENTED** [review]
Hi @sven1103 I did not have the same result as you. For me, the style looked as Steffen showed in his screenshot and it was also responsive to resizing.

The only thing that we might want to adapt is:
when resizing to the smalles possible size there is no spacing left between font/buttons and the screen. But since the screen will never be that small we can also neglect it for now.

File: vaadinfrontend/src/main/java/life/qbic/datamanager/views/components/CardLayout.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+import com.vaadin.flow.component.orderedlayout.FlexComponent;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+
+/**
+ * <b>Card Layout</b>
+ *
+ * <p>A card Component with a shadow containing customizable components such as a title, a layout
+ * for fields and a layout for buttons.
+ *
```
Comment (sven1103) [inline vaadinfrontend/src/main/java/life/qbic/datamanager/views/components/CardLayout.java:16]: Maybe a little bit more information about the different layouts:

As a user of this class, what do I need to know to format it? What can I expect the layout to look like? When I extend the class, that do I need to be aware of?

**Comment by sven1103** [comment]
Due to time-pressure, I approve this know, since it is a blocking PR. 

@Steffengreiner can you please extend the Java Docs later and add a section in the README for developers, so this re-usable class gets awareness.

Thanks a lot!

**Review by sven1103 — APPROVED** [review]
Approved, with the emphasis on more extensive documentation for the CardLayout.class

**Pull request merged**